src/dqe_idx.erl

```diff
@@ -227,6 +228,28 @@ metrics(Collection) ->
 
 %%--------------------------------------------------------------------
 %% @doc
+%% Returns a list of metric path suffixes of `Depth' that are prefixed
+%% by the given probe `Prefix', which can also be empty.
+%%
+%% For example:
+%% metrics(<<"collection">>, <<>>, 1) -> [<<"base">>].
+%% metrics(<<"collection">>, <<"base">>, 1) -> [<<"cpu">>].
+%% metrics(<<"collection">>, <<>>, 2) -> [[<<"base">>.<<"cpu">>]].
+%%
+%% Actual results are binary-encoded in `dproto' encoding.
+%% @end
+%%--------------------------------------------------------------------
+-spec metrics(Collection::collection(),
+              Prefix::metric(),
```

Comment:
Why is `Prefix` a `metirc()` not a `[metric()]` this would make it impossible to use a prefix more then 1 way down. Say for example 1st querying `([<<"base">>], 1)` then from there drilling down to `([<<"base">>, <<"cpu">>], 1)` and so forth.

I had the impression to allow drill down exploration was the main aim of the metric_variants (or now metrics) function?